internal/proxy/look_aside_balancer.go

```diff
@@ -263,11 +263,25 @@ func (b *LookAsideBalancer) trySetQueryNodeUnReachable(node int64, err error) bo
 	failures.Inc()
 	b.failedHeartBeatCounter.Insert(node, failures)
 
+	log.Info("get component status failed",
+		zap.Int64("node", node),
+		zap.Int64("times", failures.Load()),
+		zap.Error(err))
+
 	if failures.Load() < Params.ProxyCfg.RetryTimesOnHealthCheck.GetAsInt64() {
-		log.Warn("get component status failed",
-			zap.Int64("node", node),
-			zap.Int64("times", failures.Load()),
-			zap.Error(err))
+		return false
+	}
+	// if the total time of consecutive heartbeat failures reach the session.ttl, remove the offline query node
+	limit := Params.CommonCfg.SessionTTL.GetAsDuration(time.Second).Seconds() /
+		Params.ProxyCfg.HealthCheckTimeout.GetAsDuration(time.Millisecond).Seconds()
+	if failures.Load() > Params.ProxyCfg.RetryTimesOnHealthCheck.GetAsInt64() && float64(failures.Load()) >= limit {
```

Comment:
NIT: 
```suggestion
	if failures.Load() >= Params.ProxyCfg.RetryTimesOnHealthCheck.GetAsInt64() && float64(failures.Load()) >= limit {
```

Reply:
after query node become unreachable, it needs at least one more heartbeat to make it unreachable, in case of nodes has been removed before it become unreachable